Add Multi-Issuer docs page (#1176)
DuendeSoftware/docs.duendesoftware.com: astro/src/content/docs/identityserver/tokens/issuer.mdx

---
title: "Issuer Claim"
description: "Overview of the issuer claim in a JWT and how IdentityServer can serve different iss claims."
date: 2026-08-03T08:22:12+02:00
sidebar:
  order: 210
---

## Issuer Claim

JWT [RFC7519](https://datatracker.ietf.org/doc/html/rfc7519#section-4.1.1) defines the Issuer Claim `iss`, which identifies the principal that issued the JWT. The claim holds the unique identifier for the Authorization Server. When a service receives a JWT, it should verify that the JWT came from the expected Authorization Server. It does so using the `iss` claim.

> IdentityServer inserts the `iss` claim value using the origin IdentityServer is hosted from. If [mTLS](/identityserver/tokens/mtls-setup.mdx) is configured with a valid domain name, part of that domain will also be included. You can choose to override this dynamic behavior by setting a static value for the `IssuerUri` property in [IdentityServer Options](/identityserver/reference/v8/options).

Reviewer: The issuer should always be equal to the URI where the discovery document is hosted. So in an MTLS setup using a subdomain (e.g. `mtls.auth.example.org` for issuer `auth.example.org`), the issuer claim will read `auth.example.org`. When configuring MTLS using a FQDN (e.g. `mtls.example.org`), then you're required to configure the static `IssuerUri` property to `auth.example.org`.

I would leave out the MTLS part here though, since it shouldn't influence the issuer claim value.

```suggestion
IdentityServer inserts the `iss` claim value using the origin IdentityServer is hosted from by default. You can change this behavior by setting the `IssuerUri` property in [IdentityServer Options](/identityserver/reference/v8/options).
```

## Multi-Issuer

> The IdentityServer Multi-Issuer feature allows a single IdentityServer to respond with different `iss` claim values depending on the domains it hosts.

Reviewer 2: ```suggestion
Duende IdentityServer can respond with different `iss` claim values depending on the domains it hosts.
```

(being preemptive about rename of the add-on)

Imagine a single IdentityServer instance is hosted over 3 different domains like `auth.dailyreader.example`, `auth.streamwave.example`, and `auth.podvault.example`. If that instance of IdentityServer was developed to allow users to sign-in across all three domains with the same credentials, the result token still only works for the APIs accepting tokens from the target issuer. For example, if the backend API only allows tokens with the issuer `auth.dailyreader.example`, a user's token with `iss` claim `auth.streamwave.example` will not be accepted. Even though their credentials work across all three domains, the token is limited to the one issuer.

> If you set a static value for the `IssuerUri` property in [IdentityServer Options](/identityserver/reference/v8/options), the Multi-Issuer feature will not be enabled because the returned `iss` claim will always be the value you set.

Reviewer: ```suggestion
If you configure the `IssuerUri` property in [IdentityServer Options](/identityserver/reference/v8/options), the Multi-Issuer feature will not be enabled: the returned `iss` claim will always contain the configured `IssuerUri` value.
```

> If you set a static value for the `IssuerUri` property in [IdentityServer Options](/identityserver/reference/v8/options), the Multi-Issuer feature will not be enabled because the returned `iss` claim will always be the value you set.

Reviewer 2: ```suggestion
If you set a static value for the `IssuerUri` property in [IdentityServer Options](/identityserver/reference/v8/options), Multi-Issuer will not be enabled because the returned `iss` claim will always be the value you set.
```
